libs/datasets/dataset_pointer.py

```diff
@@ -51,7 +53,32 @@ def save_dataset(self, dataset: DatasetBase) -> pathlib.Path:
         _logger.info("Successfully saved dataset", path=str(self.path))
         return self.path
 
-    def load_dataset(self) -> DatasetBase:
+    def load_dataset(
+        self, before: str = None, previous_commit: bool = False, commit: str = None
+    ) -> DatasetBase:
+        """Load dataset from file specified by pointer.
+
+        If options are specified, will load from git history of the file.
+
+        Args:
+            before: If set, returns dataset from first commit for file before date.
+            previous_commit: If true, returns the dataset from previous commit.
+            commit: SHA of specific commit.
+
+        Returns: Instantiated dataset.
+        """
+        if before or previous_commit or commit:
+            # When loading dataset from a previous point in time, the data is not available as path.
+            # To get around this, we save the bytes of the commit data to a temporary file to then
+            # load as a csv.
+            with tempfile.NamedTemporaryFile() as tmp_file:
+                path = pathlib.Path(tmp_file.name)
+                lfs_data = git_lfs_object_helpers.get_data_for_path(
+                    self.path, before=before, previous_commit=previous_commit, commit=commit
+                )
+                path.write_bytes(lfs_data)
+                return self.dataset_type.dataset_class.load_csv(path)
```

Comment:
can you change TimeseriesDataset.load_csv to accept `path_or_buf: Union[pathlib.Path, TextIO]` like common_df.read_csv ? Then I think you'll be able to keep the data in memory without using a temp file.